### ISSUE TITLE
How to center camera on user location?

### ISSUE THREAD

**Issue by AnEpicName** [mapbox/mapbox-maps-flutter#33]
I'm trying to get my location from the map to avoid using external libraries like `geolocator`, but I can't find any option nor listener for the user location.

At the start of my view I want to center the camera at the current user location. Is there any way to do that in the current version of the plugin?

**Comment by kamami** [comment]
Have a look at this example: https://gist.github.com/hkben/844d46a829f667e91fa6e6078e8b019c
Maybe it is helpful

**Comment by AnEpicName** [comment]
Thanks! This is exactly what I was looking for!

**Comment by geisterfurz007** [comment]
I'd argue that there is a case for this being directly supported by this library since both the Android SDK and the iOS SDK support following the puck out of the box: [https://docs.mapbox.com/android/maps/guides/user-location/#location-tracking](Android Docs) | [https://docs.mapbox.com/ios/maps/guides/user-location/#location-tracking](iOS Docs)

**Comment by kamami** [comment]
But viewport is not supported by the Flutter SDK yet.

**Comment by geisterfurz007** [comment]
@kamami Yeah, that's what I mean; I think this issue should remain open until this is supported in the Flutter SDK.

**Comment by serraojoao** [comment]
@kamami you refer to this package support for viewport or flutter framework itself? 

This would be a good improvement as when using the timer approach to center the location we notice the map "jumping" every second instead of the smooth navigation when using mapbox in a native app.

**Comment by yunikkk** [comment]
@geisterfurz007 @kamami @serraojoao reopening to track when viewport will be implemented. 
There're no estimations when it'll become available yet, I'll keep you all updated.

**Comment by basvdijk** [comment]
@yunikkk I understand you can't give an estimate. But is there a ballpark figure you can say like if it takes some months, more than half a year, a year? This would really help for my project planning.

**Comment by yunikkk** [comment]
> @yunikkk I understand you can't give an estimate. But is there a ballpark figure you can say like if it takes some months, more than half a year, a year? This would really help for my project planning.

@evil159 or @baleboy - do you folks know if the viewport API is planned anytime soon?

**Comment by baleboy** [comment]
Unfortunately there is no committed timeline at the moment.

**Comment by MostafaSolimanMO** [comment]
Any update? :)

**Comment by maios** [comment]
Hi, we have added Viewport support since mapbox_maps_flutter 2.5.0
To center camera on user's location, you can use [FollowPuckViewportState](https://pub.dev/documentation/mapbox_maps_flutter/latest/mapbox_maps_flutter/FollowPuckViewportState-class.html)
